business_requirement_earned_value/report/project_completion_report.py

```diff
@@ -0,0 +1,224 @@
+    )
+    task_category_id = fields.Many2one(
+        'project.category', string='Task Category'
+    )
+    project_state = fields.Char(
+        'Project state', readonly=True, help="Project State")
+    br_kanban_state = fields.Selection(
+        [
+            ('normal', 'In Progress'),
+            ('on_hold', 'On Hold'),
+            ('done', 'Ready for next stage')
+        ],
+        'BR Kanban State',
+        required=False,
+        default='normal'
+    )
+    task_kanban_state = fields.Selection([
+        ('normal', 'In Progress'),
+        ('done', 'Ready for next stage'),
+        ('blocked', 'Blocked')
+    ], string='Activity Kanban State',
+    )
+    priority = fields.Selection([('0', 'Low'), ('1', 'Normal'),
+                                 ('2', 'High')],
+                                'Priority', readonly=True)
+    date_deadline = fields.Date(string='Deadline')
+
+    @api.model_cr
+    def init(self):
+        """
+            Project Completion Report
+        """
+        tools.drop_view_if_exists(self._cr, 'project_completion_report')
+        self._cr.execute("""
+                CREATE OR REPLACE VIEW project_completion_report AS (
+                    -- Since Odoo requires a unique ID for each line and since
+                    -- some issues and tasks might share the same ID, use the
+                    -- row number to ensure each row has a unique ID
+                    SELECT
+                        row_number() OVER (ORDER BY q.activity_id) AS id, q.*
+                    FROM
+                    (
+                        (
+                            SELECT
+                                a.partner_id,
+                                b.project_id AS master_project_id,
+                                b.id AS br_id,
+                                b.state AS br_status,
+                                t.categ_id AS task_category_id,
+                                CASE WHEN p.active = True THEN 'Active'
+                                WHEN p.active=False THEN 'Archived'
+                                END AS project_state,
+                                p.id AS project_id,
+                                a.id AS account_id,
+                                'task' AS activity_type,
+                                t.id AS activity_id,
+                                t.name AS activity_name,
+                                t.user_id,
+                                t.stage_id AS activity_stage_id,
+                                CASE WHEN r.uom_id=(
+                                select res_id from ir_model_data
+                                where name='product_uom_hour')
+                                THEN COALESCE(r.qty, 0)
+                                WHEN r.uom_id=(
+                                select res_id from ir_model_data where
+                                name='product_uom_day')
+                                 THEN COALESCE(r.qty*8, 0)

```

Comment:
Your solution works for hours and days (which is much better than before and solves 90% of the cases) but what if we use weeks or months?
Isnot there a better way to get the factor of the uom and use it for the calculation instead of hard-coded "8"
It would be more generic and would not crash in case of specific UoM.
cc @lonelysun